### PR TITLE
Oliinyk/Supporting edge case when stepping next in workshop form while address invalid and collapsed

### PR DESCRIPTION
**Pull request by doliinyk**
https://github.com/ita-social-projects/OoS-Frontend/issues/3035

## Summary by CodeRabbit

* **Bug Fixes**
  * Form validation now scrolls to and focuses only the first actually visible invalid field (ignores hidden or zero-size elements and those inside hidden parents), reducing unexpected jumps and improving accessibility.
  * When a submit handler is provided, the submission path is used instead of advancing the stepper.

* **UI**
  * Removed a stray character in a button to prevent an unintended visible text node.

* **Tests**
  * Added edge-case coverage for element visibility and expanded submit vs. next-step behavior tests.

### CODE REVIEW

File: src/app/shared/directives/stepper/stepper.directive.ts [hunk 1]
```diff
@@ -59,9 +59,35 @@ export class StepperDirective {
...
+        if (firstVisibleInvalidField) {
+          firstVisibleInvalidField.scrollIntoView({ behavior: 'smooth', block: 'center' });
+          (firstVisibleInvalidField as HTMLElement).focus({ preventScroll: true });
+        }
       }
     });
   }
+
+  private isElementVisible(element: HTMLElement): boolean {
+    const computedStyle = window.getComputedStyle(element);
```
Comment (humanmantis) [inline src/app/shared/directives/stepper/stepper.directive.ts]: Please use WINDOW injection token instead of working directly with the Window object.

**Review by humanmantis — CHANGES_REQUESTED** [review]

**Review by humanmantis — APPROVED** [review]
